tests/client/components/carousel.spec.js

```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
+    it.skip('should display the second slide after waiting the timeout duration with direction forward', () => {});
+
+    // start on the second slide and allow autoplay to advanced to the first slide
+    it.skip('should display the first slide after waiting the timeout duration with direction backward', () => {});
+
+    it.skip('should not allow the autoplay feature to advance on hover', () => {});
+
+    it.skip('should allow the autoplay feature to advance on hover when hover pause false', () => {});
 
-    expect(wrapper.vm.canAdvanceBackward).toBeFalsy();
+    it.skip('should show the first slide twice when looping is enabled with autoplay', () => {});
 
-    expect(wrapper).toMatchSnapshot();
+    it('should reset autoplay when switching slide without autoplayHoverPause', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          perPage: 1,
+          autoplay: true,
+          autoplayHoverPause: false
+        },
+        slots: {
+          default: [Slide, Slide]
+        }
+      });
+
+      const spy = jest.spyOn(wrapper.vm, 'restartAutoplay');
+      wrapper.vm.goToPage(2);
+      expect(wrapper.vm.restartAutoplay).toHaveBeenCalled();
+      spy.mockRestore();
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
+
+    it('should not reset autoplay when switching slide with autoplayHoverPause', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          perPage: 1,
+          autoplay: true,
+          autoplayHoverPause: true
+        },
+        slots: {
+          default: [Slide, Slide]
+        }
+      });
+
+      const spy = jest.spyOn(wrapper.vm, 'restartAutoplay');
+      wrapper.vm.goToPage(2);
+      expect(wrapper.vm.restartAutoplay).not.toHaveBeenCalled();
+      spy.mockRestore();
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
   });
 
-  it('should be unable to advance forward by default (no slides added)', () => {
-    const wrapper = mount(Carousel);
+  describe('Center mode', () => {
+    it.skip('should not center image when image is less than the container width', () => {});
 
-    expect(wrapper.vm.canAdvanceForward).toBeFalsy();
+    it.skip('should center image when image is less than container width and centerMode is true', () => {});
+  });
+
+  describe('Custom easing property', () => {
+    it.skip('should set easing to ease by default', () => {});
 
-    expect(wrapper).toMatchSnapshot();
+    it.skip('should set easing to the custom easing supplied', () => {});
+
+    // TODO: Should we be testing for valid CSS transition easing like the README suggests?
   });
 
-  it('should apply custom slides per page when responsive param provided', () => {
-    const wrapper = mount(Carousel, {
-      propsData: {
-        perPageCustom: [[0, 20]]
-      }
+  describe('Dynamic slide count', () => {
```

Comment:
Not 100% here but I think we can also test tabIndex of generated slides here

Reply:
Added tests for this